lib/ddtrace/contrib/sequel/patcher.rb

```diff
@@ -0,0 +1,56 @@
+require 'ddtrace/contrib/sequel/database'
+require 'ddtrace/contrib/sequel/dataset'
+
+module Datadog
+  module Contrib
+    module Sequel
+      # Patcher enables patching of 'sequel' module.
+      # This is used in monkey.rb to manually apply patches
+      module Patcher
+        include Base
+
+        APP = 'sequel'.freeze
+
+        register_as :sequel, auto_patch: false
+        option :service_name
+        option :tracer, default: Datadog.tracer
+
+        @patched = false
+
+        module_function
+
+        # patched? tells whether patch has been successfully applied
+        def patched?
+          @patched
+        end
+
+        def patch
+          if !@patched && compatible?
+            begin
+              patch_sequel_database
+              patch_sequel_dataset
+
+              @patched = true
+            rescue StandardError => e
+              Datadog::Tracer.log.error("Unable to apply Sequel integration: #{e}")
+            end
+          end
+
+          @patched
+        end
+
+        def compatible?
+          defined?(::Sequel)
```

Comment:
shouldn't we check also the Ruby version? I think we should simply say that we can't patch Sequel rather then trying and raising an exception.

Reply:
Ah, great idea, I'll do that.